src/kokkos_ext/ArborX_DetailsKokkosExtStdAlgorithms.hpp

```diff
@@ -149,6 +150,49 @@ void iota(ExecutionSpace const &space, ViewType const &v,
       KOKKOS_LAMBDA(int i) { v(i) = value + (ValueType)i; });
 }
 
+template <typename Iterator>
+KOKKOS_FUNCTION void nth_element(Iterator first, Iterator nth, Iterator last)
```

Comment:
I agree with Daniel that this algorithm is not a good fit in that header, next to other algorithms that act on views.  I would prefer if you introduced a new header that have minimal includes.

Reply:
> I would prefer if you introduced a new header that have minimal includes.

I introduced (a long winded) `ArborX_DetailsKokkosExtKernelStdAlgorithms.hpp`. My argument for that is that it is still a `std::` algorithm.